.github/workflows/test.yaml

```diff
@@ -170,6 +175,57 @@ jobs:
       - name: Report
         id: report
         run: echo "outcome=$([ ${{ steps.metadataYaml.outputs.exists }} = 'true' ] && echo success || echo skip)" >> $GITHUB_OUTPUT
+  tox-lint:
+    name: Lint tox.ini
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    outputs:
+      outcome: ${{ steps.report.outputs.outcome }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run lint
+        run: |
+          # Check dependencies
+          EXPECTED_LINT_DEPS="\
+              mypy \
+              isort \
+              black \
+              flake8-docstrings \
+              flake8-docstrings-complete \
+              flake8-builtins \
+              flake8-test-docs \
+              pep8-naming \
+              codespell \
+              pylint \
+              pydocstyle \
+              "
+          for EXPECTED_LINT_DEP in $EXPECTED_LINT_DEPS; do
+              DEP_REGEX="deps.*=.*\n(\s.*\n)*(\s.*$EXPECTED_LINT_DEP)"
+              python3 -c "from pathlib import Path; import re; import sys; sys.exit(1) if not re.search(r'$DEP_REGEX', Path('tox.ini').read_text()) else sys.exit(0)" || \
```

Comment:
Not a big fan of mixing bash and inline python, wouldn't it be better either to do the whole step in python, or use grep here ?

Reply:
Now that it is based on the command output, `grep` is used